internal/bminventory/inventory.go

```diff
@@ -3231,10 +3245,9 @@ func (b *bareMetalInventory) processDiskSpeedCheckResponse(ctx context.Context,
 		if diskPerfCheckResponse.IoSyncDuration > thresholdMs {
 			// If the 99th percentile of fdatasync durations is more than 10ms, it's not fast enough for etcd.
 			// See: https://www.ibm.com/cloud/blog/using-fio-to-tell-whether-your-storage-is-fast-enough-for-etcd
-			msg := fmt.Sprintf("Host's disk %s is slower than the supported speed, and may cause degraded cluster performance (fdatasync duration: %d ms)",
-				diskPerfCheckResponse.Path, diskPerfCheckResponse.IoSyncDuration)
-			log.Warnf(msg)
-			b.eventsHandler.AddEvent(ctx, *h.ClusterID, h.ID, models.EventSeverityWarning, msg, time.Now())
+			ev := eventgen.NewSlowHostDiskEvent(*h.ClusterID, *h.ID, diskPerfCheckResponse.Path, diskPerfCheckResponse.IoSyncDuration)
+			b.eventsHandler.SendHostEvent(ctx, ev, time.Now())
+			log.Warnf(ev.FormatMessage())
```

Comment:
pls move one line above to preserve the original order (log before event)

Reply:
solved.
does logs before actions is a convention?

Reply:
no, there are other places in which the order is reversed.
but in this PR we'd like to preserve same behavior as used to be before.
if someone would decide that such a change is required, that can be added in a separate PR which isn't a concern of the the current one.